File: UNXNavigatorDemo/Feature/Basic/Controllers/ViewController09_FullScreen.m
//
//  ViewController10_FullScreen.m
//  UNXNavigatorDemo
//
//  Created by Leo Lee on 2020/10/26.
//

#import <UNXNavigator/UNXNavigator.h>

#import "ViewController09_FullScreen.h"

@interface ViewController09_FullScreen ()

@end

@implementation ViewController09_FullScreen

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.view.backgroundColor = self.randomColor;
}

- (UIStatusBarStyle)preferredStatusBarStyle {
    return UIStatusBarStyleLightContent;
}

- (UIColor *)unx_barTintColor {
    return [UIColor whiteColor];
}

- (NSDictionary<NSAttributedStringKey,id> *)unx_titleTextAttributes {
    return @{NSForegroundColorAttributeName: [self unx_barTintColor]};
}

- (UIColor *)unx_navigationBarBackgroundColor {
    return self.randomColor;
}

- (UIColor *)unx_shadowImageTintColor {
    return [UIColor clearColor];
}

@end

File: Example/UNXNavigatorSwiftDemo/Basic/ViewController09_FullScreen.swift
//
//  ViewController09_FullScreen.swift
//  UNXNavigatorSwiftDemo
//
//  Created by Leo Lee on 2020/11/23.
//

import UIKit
import UNXNavigator

class ViewController09_FullScreen: BaseViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        view.backgroundColor = randomColor
    }
    
    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent
    }
    
    override var unx_barTintColor: UIColor? {
        return .white
    }
    
    override var unx_titleTextAttributes: [NSAttributedString.Key : Any]? {
        return [NSAttributedString.Key.foregroundColor: UIColor.white]
    }
    
    override var unx_navigationBarBackgroundColor: UIColor? {
        return randomColor
    }
    
    override var unx_shadowImageTintColor: UIColor? {
        return .clear
    }

}
